### ISSUE TITLE
Relogin without loss of user-specific settings (especially EXIF tags)

### ISSUE THREAD

**Issue by OpenGreenStreet** [commons-app/apps-android-commons#5766]
### What is the user problem or growth opportunity you want to see solved?

If a user changes their password, the Commons app will not be notified. It is then no longer possible to upload images. To upload images again, the user must log out and log in again. The user-specific settings (especially EXIF tags) will be lost. Relogin without losing user-specific settings (especially EXIF tags)

### How do you know that this problem exists today? Why is this important?

 The user-specific settings (especially EXIF tags) will be lost. 

### Who will benefit from it?

All users

**Comment by sivaraam** [comment]
We could try to solve this by giving a general setting export/import option. The relogin would help your specific case but a settings export/import option would be more helpful as it would generally give an option to restore the settings, say, when the user switches to a new device.

**Comment by psh** [comment]
Other apps sometimes have an option for "logout and remember my details" verses "logout and forget me".  It seems like a lot of work for users to have to remember to export when it could be a simple option as part of logout.

**Comment by OpenGreenStreet** [comment]
@psh Exactly, you can get by with it if you check the settings after logging in again. But it would be very convenient if the app did this automatically.

**Comment by sivaraam** [comment]
Got it. So, I think a option like "Remember my settings" during logout is what we feel would be beneficial. Does that seem right?

I think we could the settings export/import migjt still be helpful for other use cases.

**Comment by OpenGreenStreet** [comment]
@sivaraam : yes, "Remember my settings" would be very nice

**Comment by parneet-guraya** [comment]
https://github.com/commons-app/apps-android-commons/blob/0153cbe0ed92f2ca1a6685798e5f49bd04fe967c/app/src/main/java/fr/free/nrw/commons/CommonsApplication.kt#L220

We're clearing preferences on logout. There are two types of logout events where this preference is cleared ->

- Intentional, when user click on logout.
- Unintentional, when we get `InvalidLoginTokenException` and app logs out. Don't know when this occurs maybe when user change password from other client?

I think we should prompt the user when there's `Intentional` log out and ask in a checkbox if want to remember the settings or not.

In case of `Unintentional` log outs user may expect their settings to be preserved.

@OpenGreenStreet What do you think?

**Comment by nicolas-raoul** [comment]
> I think we should prompt the user when there's Intentional log out and ask in a checkbox if want to remember the settings or not.
> In case of Unintentional log outs user may expect their settings to be preserved.

Sounds reasonable to me! :-)

**Comment by OpenGreenStreet** [comment]
@parneet-guraya : Please excuse the late reply:

* "In case of Unintentional log outs user may expect their settings to be preserved."
** That is exactly the point. Every time I had a problem, it was because I had changed my password with the Wikipedia web interface. The next time I used the Commons app, I didn't remember this and had problems uploading images. There was no meaningful hint.
It would be important to me that all settings are retained in this case.

**Comment by parneet-guraya** [comment]
Okay, so we can have feature for storing/restoring settings or preferences. Then it would depend when it will work based on 
- If `Unintentional` logout -> always enabled.
- If `Intentional` logout -> ask user if want to remember settings or not.

@nicolas-raoul  Can I start working on this? :-)

**Comment by nicolas-raoul** [comment]
Is there any drawback to having it alwayp enabled, even for intentional logouts? Users who want to wipe their settings can use Android's UI for that. 🙂
